avocado/plugins/spawners/process.py

```diff
@@ -42,18 +42,8 @@ async def wait_task(runtime_task):
 
     @staticmethod
     async def check_task_requirements(runtime_task):
-        runnable_requirements = runtime_task.task.runnable.requirements
-        if not runnable_requirements:
-            return True
-
-        for requirements in runnable_requirements:
-            for (req_type, req_value) in requirements.items():
-                # The fact that this is avocado code means this
-                # requirement is fulfilled
-                if req_type == 'core' and req_value == 'avocado':
-                    continue
-                else:
-                    # current implementation can not check any other type of
-                    # requirement at this moment so fail
-                    return False
+        """Check for requirements a runtime task may have to run."""
+        # right now, checks only if the runner is available.
+        if runtime_task.task.runnable.pick_runner_command() is None:
```

Comment:
Indeed, Avocado should have some mechanism to fulfill internal requirements. As an internal requirement, I don't see the need for a user to specify it in the test itself as a test requirement, but we should have a way to check those requirements depending on the spawner the test runs on.

I could not find a way to fit the current code in the Requirements Resolver structure because the Requirements Resolver is supposed to handle test requirements and not internal requirements. I see a way to implement internal requirements and much more in the proposal I made here https://github.com/avocado-framework/avocado/discussions/4552#discussioncomment-671334.

Another downside I see with the current code is the increase of `if` statements with the increase of tags that need to be verified in the spawner's code. Again, I think it is important to check the spawner's requirements, but not as a test requirement declared in the test code. This is the main reason I opted for removing it for now. I know it still exists in the `podman` spawner and my suggestion is to remove it for now and re-introduce it in a broader dependency structure as I proposed.